earthaccess/results.py

```diff
@@ -107,6 +109,41 @@ def get_umm(self, umm_field: str) -> Union[str, Dict[str, Any]]:
         """
         return self["umm"].get(umm_field, "")
 
+    def doi(self) -> str | None:
+        """Retrieve the Digital Object Identifier (DOI) for this collection.
+
+        Returns:
+            This collection's DOI information, or `None`, if it has none.
+        """
+        doi = self["umm"].get("DOI", {})
+        if isinstance(doi, dict):
+            return doi.get("DOI", None)
+        return None
+
+    def citation(self, *, format: str, language: str) -> str | None:
```

Comment:
Let's cache the citations to avoid making multiple requests for the same citation:

```suggestion
    @cache
    def citation(self, *, format: str, language: str) -> str | None:
```

Reply:
Ah, right, because `DataCollections` is a `dict`.

In that case, let's create a top-level function in this file, and we'll decorate that with `@cache`, and have this method call the function, like so:

```python
@cache
def _citation(*, doi: str, format: str, language: str) -> str:
    response = requests.get(
        "https://citation.doi.org/format",
        params={"doi": doi, "style": format, "lang": language},
    )
    response.raise_for_status()
    return response.text
```

Please copy the docstring from the `citation` method to the new `_citation` function above, and adjust it accordingly since the `doi` is now a parameter, and the return type is `str` instead of `str | None` since `doi` is `str` instead of `str | None`.

Then, call it from the `citation` method, like so (keeping the current docstring as is):

```python
def citation(self, *, format: str, language: str) -> str | None:
    return (
        None
        if (doi := self.doi()) is None
        else _citation(doi=doi, format=format, language=language)
    )
```